nexus/tests/integration_tests/silos.rs

```diff
@@ -2437,3 +2440,85 @@ async fn check_fleet_privileges(
             .unwrap();
     }
 }
+
+// Test that a silo admin can create new certificates for their silo
+//
+// Internally, the certificate validation check requires the `authz::DNS_CONFIG`
+// resource (to check that the certificate is valid for
+// `{silo_name}.{external_dns_zone_name}`), which silo admins may not have. We
+// have to use an alternate, elevated context to perform that check, and this
+// test confirms we do so.
+//
+// Ensures no regression on
+// https://github.com/oxidecomputer/omicron/issues/4532.
```

Comment:
Personally I don't bother referencing past bugs -- I think it's enough to explain why this is useful/necessary for correctness.